library/index.c

```diff
@@ -323,24 +323,25 @@ int eblob_index_blocks_fill(struct eblob_base_ctl *bctl)
 }
 
 
-static struct eblob_disk_control *eblob_find_on_disk(struct eblob_backend *b,
-		struct eblob_base_ctl *bctl, struct eblob_disk_control *dc,
+static int eblob_find_on_disk(struct eblob_backend *b,
```

Comment:
There are multiple pread() calls in this function, there should be only one right after eblob_index_blocks_search_nolock() like the one at 378 line.

There should be no other reads from index file descriptor like at lines 428 and 472.
Looks like you do not use information obtained from range index tree and reads index data outside of the range potentially covering needed key.